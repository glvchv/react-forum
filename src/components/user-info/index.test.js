import React from 'react';
import renderer from 'react-test-renderer';
import UserInfo from '.';
import { BrowserRouter } from 'react-router-dom';

jest.mock('../minified-post', () => 'MinifiedPost');

const user = {
    _id: '5f38e64772f6952ab4fae3f0',
    username: 'vld175',
    posts: [{ title: '0' }, { title: '1' }, { title: '2' }]
}

describe('User-info component', () => {
    it('should render user\'s own profile', () => {
        localStorage.setItem('userId', '5f38e64772f6952ab4fae3f0');
        const tree = renderer.create(
            <BrowserRouter>
                <UserInfo user={user} />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render another user\'s profile', () => {
        localStorage.setItem('userId', '123');
        const tree = renderer.create(
            <BrowserRouter>
                <UserInfo user={user} />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});