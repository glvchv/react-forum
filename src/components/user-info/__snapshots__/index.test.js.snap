// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User-info component should render another user's profile 1`] = `
<div
  className="wrapper"
>
  <div
    className="avatar-section"
  >
    <img
      alt="avatar"
      className="avatar"
      src="/default.png"
    />
    
  </div>
  <div
    className="personal-section"
  >
    <div
      className="leftside"
    >
      <p>
        <small>
          Username: 
        </small>
        vld175
      </p>
    </div>
    <div
      className="rightside"
    >
      <p>
        <small>
          Posted: 
        </small>
        3
         times
      </p>
    </div>
  </div>
  <h1
    className="centered"
  >
    Posts: 
  </h1>
  <div
    className="posts-wrapper"
  >
    <MinifiedPost
      post={
        Object {
          "title": "0",
        }
      }
    />
    <MinifiedPost
      post={
        Object {
          "title": "1",
        }
      }
    />
    <MinifiedPost
      post={
        Object {
          "title": "2",
        }
      }
    />
  </div>
</div>
`;

exports[`User-info component should render user's own profile 1`] = `
<div
  className="wrapper"
>
  <div
    className="avatar-section"
  >
    <img
      alt="avatar"
      className="avatar"
      src="/default.png"
    />
    
  </div>
  <div
    className="personal-section"
  >
    <div
      className="leftside"
    >
      <p>
        <small>
          Username: 
        </small>
        vld175
      </p>
    </div>
    <div
      className="rightside"
    >
      <p>
        <small>
          Posted: 
        </small>
        3
         times
      </p>
    </div>
  </div>
  <h1
    className="centered"
  >
    Posts: 
  </h1>
  <div
    className="posts-wrapper"
  >
    <MinifiedPost
      post={
        Object {
          "title": "0",
        }
      }
    />
    <MinifiedPost
      post={
        Object {
          "title": "1",
        }
      }
    />
    <MinifiedPost
      post={
        Object {
          "title": "2",
        }
      }
    />
  </div>
</div>
`;
